modules/imap_folders/site.js

```diff
@@ -1,51 +1,79 @@
 'use strict';
 
-var folder_page_folder_list = function(container, title, link_class, target, id_dest, subscription = false) {
+var folder_page_folder_list = function (container, title, link_class, target, id_dest, subscription = false) {
     var id = $('#imap_server_folder').val();
-    var folder_location = $('.'+container);
-    $('li', folder_location).not('.'+title).remove();
-    var folders = $('.folder_list .imap_'+id+'_').clone(false);
+    var folder_location = $('.' + container);
+    $('li', folder_location).not('.' + title).remove();
+    var folders = $('.folder_list .imap_' + id + '_').clone(false);
     folders.find('.manage_folders_li').remove();
     $('.imap_folder_link', folders).addClass(link_class).removeClass('imap_folder_link');
     folder_location.prepend(folders);
     folder_location.show();
-    $('.'+link_class, folder_location).on("click", function() { return expand_folders_page_list($(this).data('target'), container, link_class, target, id_dest, subscription); });
-    $('a', folder_location).not('.'+link_class).not('.close').off('click');
-    $('a', folder_location).not('.'+link_class).not('.close').on("click", function() { set_folders_page_value($(this).data('id'), container, target, id_dest); return false; });
-    $('.close', folder_location).on("click", function() {
+
+    // 1. Get the <a> element
+    const link = document.querySelector('.'+link_class);
+    // 2. Save the <i> element
+    const original_icon = link.querySelector('i');
+    const original_icon_clone = original_icon.cloneNode(true);
+    // 3. Create the spinner
+    const spinner = document.createElement('div');
```

Comment:
Why do you have all these unrelated updates when the PR is for sieve connection? It is better to have PRs that are self-contained - easier to review, easier to trace in the history.